src/java_bytecode/java_object_factory.cpp

```diff
@@ -1019,9 +1030,12 @@ void gen_nondet_init(
   //   <type> <identifier>;
   for(const symbolt * const symbol_ptr : symbols_created)
   {
-    code_declt decl(symbol_ptr->symbol_expr());
-    decl.add_source_location()=loc;
-    init_code.add(decl);
+    if(!symbol_ptr->is_static_lifetime)
+    {
+      code_declt decl(symbol_ptr->symbol_expr());
```

Comment:
Test not possible because if there's only ever a `setstatic` instruction then the thing I'm testing cannot be external (that would have to be retrieved with `getstatic`)